lib/cylc/review.py

```diff
@@ -78,8 +79,13 @@ def __init__(self, *args, **kwargs):
         if self.host_name and "." in self.host_name:
             self.host_name = self.host_name.split(".", 1)[0]
         self.cylc_version = CYLC_VERSION
-        template_env = jinja2.Environment(loader=jinja2.FileSystemLoader(
-            get_util_home("lib", "cylc", "cylc-review", "template")))
+        template_env = jinja2.Environment(
+            autoescape=select_autoescape(
+                enabled_extensions=('rc'),
+                default_for_string=False,
+                default=True),
+            loader=jinja2.FileSystemLoader(
+                get_util_home("lib", "cylc", "cylc-review", "template")))
```

Comment:
These changes looks fine, but has anyone tested that they don't adversely affect `cylc review`?

Reply:
In #2945 I also added in an non-default (i.e. not False) `` autoescape`` to the ``jinja2.Environment()`` call, as I noticed the issue independently, i.e. without bandit's issue raising, & was not aware it was being addressed here. Sorry about that. Indeed the associated Codacy flag was only raised when I made the change to fix the security issue!

The ``autoescape=`` value applied in #2945 is different to that applied in this PR, & I verified the former solves the vulnerability (manually, as testing of markup rendering is difficult & would be too time-consuming given priorities) & also that, after adding in some Jinja2 ``safe`` filters which were required to revert some differences in displayed text, there are no adverse affects.

Is this auto-escaping variant trying to solve the same issue? If so, think I already have it covered. If not, I am unsure myself as to what other issue this is solving, or indeed what it would do, so would appreciate an brief explanation.

Reply:
this was to resolve "using jinja2 templates with autoescape=False is dangerous and can lead to XSS. Ensure autoescape=True or use the select_autoescape function to mitigate XSS vulnerabilities"

Reply:
using sadie's master version